lib/cli/src/commands/app/create.rs

```diff
@@ -1,34 +1,54 @@
 //! Create a new Edge app.
 
-use std::path::PathBuf;
-
-use anyhow::{bail, Context};
+use crate::{
+    commands::AsyncCliCommand,
+    opts::{ApiOpts, ItemFormatOpts},
+    utils::{
+        load_package_manifest,
+        package_wizard::{CreateMode, PackageType, PackageWizard},
+    },
+};
+use anyhow::Context;
 use colored::Colorize;
-use dialoguer::Confirm;
-use edge_schema::schema::StringWebcIdent;
+use dialoguer::{Confirm, Select};
 use is_terminal::IsTerminal;
-use wasmer_api::{
-    types::{DeployAppVersion, Package, UserWithNamespaces},
-    WasmerClient,
+use std::{collections::HashMap, path::PathBuf, str::FromStr};
+use wasmer_api::{types::UserWithNamespaces, WasmerClient};
+use wasmer_config::{
+    app::AppConfigV1,
+    package::{NamedPackageIdent, PackageSource, Tag},
 };
 
-use crate::{
-    commands::{
-        app::{deploy_app_verbose, AppConfigV1, DeployAppOpts, WaitMode},
-        AsyncCliCommand,
-    },
-    opts::{ApiOpts, ItemFormatOpts},
-    utils::package_wizard::{CreateMode, PackageType, PackageWizard},
-};
+use super::deploy::CmdAppDeploy;
+
+async fn write_app_config(app_config: &AppConfigV1, dir: Option<PathBuf>) -> anyhow::Result<()> {
+    let raw_app_config = app_config.clone().to_yaml()?;
+
+    let app_dir = match dir {
+        Some(dir) => PathBuf::from(dir),
+        None => std::env::current_dir()?,
+    };
+
+    let app_config_path = app_dir.join(AppConfigV1::CANONICAL_FILE_NAME);
+    std::fs::write(&app_config_path, raw_app_config).with_context(|| {
+        format!(
+            "could not write app config to '{}'",
+            app_config_path.display()
+        )
+    })
+}
 
 /// Create a new Edge app.
 #[derive(clap::Parser, Debug)]
 pub struct CmdAppCreate {
     #[clap(name = "type", short = 't', long)]
-    template: Option<AppType>,
-
+    pub template: Option<AppType>,
+    /// Whether or not to deploy the application once it is created.
+    ///
+    /// If selected, this might entail the step of publishing the package related to the
+    /// application. By default, the application is not deployed and the package is not published.
     #[clap(long)]
-    publish_package: bool,
+    pub deploy_app: bool,
```

Comment:
Should this just be `--deploy`?
The `-app` seems redundant.

Being annoying here because flags are public API and should be changed very rarely.

Reply:
@xdoardo